### PR TITLE
#993 Feature to import / export Project - crash data is skipped in se…

### PR DESCRIPTION
**Pull request by Limraj**
…rialization, error logging, correct create temp file

### CODE REVIEW

File: src/com/serotonin/mango/rt/event/type/DataSourceEventType.java [hunk 1]
```diff
@@ -122,15 +126,21 @@ public boolean equals(Object obj) {
     public void jsonSerialize(Map<String, Object> map) {
         super.jsonSerialize(map);
         DataSourceVO<?> ds = new DataSourceDao().getDataSource(dataSourceId);
-        map.put("XID", ds.getXid());
-        map.put("dataSourceEventType", ds.getEventCodes().getCode(dataSourceEventTypeId));
+        if(ds != null) {
+            map.put("XID", ds.getXid());
+            map.put("dataSourceEventType", ds.getEventCodes().getCode(dataSourceEventTypeId));
+        } else {
+            LOG.error("DataSource for id: " + dataSourceId + " is not exists.");
```
Comment (SoftQ) [inline src/com/serotonin/mango/rt/event/type/DataSourceEventType.java:133]: typo - "does not exist"

**Review by SoftQ — CHANGES_REQUESTED** [review]
please correct a typo :)

**Comment by Limraj** [comment]
The changes will still be tested.

**Comment by fabiodurao** [comment]
The export / import tool in .zip had some problems since version 1.0 of ScadaBR and remained the same in ScadaBR 1.1 (original ScadaLTS branch). The export tool in .json has a user related problem that only appeared in ScadaBR 1.1, this is probably related to the addition of user profiles that did not exist in ScadaBR 1.0.

**Comment by Limraj** [comment]
Hi @fabiodurao
has the problem you write about already described somewhere? I would be interested in the error scenario. I am happy to check if this problem occurs in the current version of the application. If the scenario has not been described yet, please create an issue. It is particularly interesting how the error manifested itself and with which function it was related to "Generate JSON" or "Export Project". 

This patch relates with bug where "Export Project" was interrupted before the process was terminated, we started the export which stopped without generate zip file.
Export crashes by incorrect data in eventhandlers, and problem with create temp file and permissions.
This patch will fix it.

export2

We have two places related to data export, this function is marked in red and green. The relationship between these functions is as follows, "Export Project" (green) uses the same mechanism to serialize data as the "Generate JSON" function, i.e. problems occurring with the "Generate JSON" function will appear in the "Export Project" function, while the "Export Project "may have bugs that arae not related to "Generate JSON". Generally "Export Project" is "Generate JSON" with all checkboxes checked, with the option to add images(graphical view), and description with app version.

Greets,
Kamil Jarmusik

**Comment by Limraj** [comment]
This topic only fix export and added api to export.

**Comment by grzesiekb** [comment]
Please add a branch to Travis so that it builds a branch with your task and creates a docker to test

**Comment by grzesiekb** [comment]
The data is exported even with a large database from a docker with tag 993. I understand that this task only includes improving the export of the project (without import). Therefore I give that it is ok.

**Review by grzesiekb — APPROVED** [review]

**Review by arturro96 — APPROVED** [review]

**Pull request merged**